libs/utils/utils.js

```diff
@@ -257,63 +257,57 @@ export function loadStyle(href, callback) {
   return loadLink(href, { rel: 'stylesheet', callback });
 }
 
-export function appendHtmlPostfix(area = document) {
+export function appendHtmlToCanonicalUrl() {
   const pageUrl = new URL(window.location.href);
   if (!pageUrl.pathname.endsWith('.html')) return;
+  const canonEl = document.head.querySelector('link[rel="canonical"]');
+  if (!canonEl) return;
+  const { href } = canonEl;
```

Comment:
I know the code is not new, but was wondering whether we really need the property;
```suggestion
  const canonUrl = new URL(canonEl.href);
  if (canonUrl.pathname.endsWith('/') || canonUrl.pathname.endsWith('.html')) return;
  const pagePath = pageUrl.pathname.replace('.html', '');
  if (pagePath !== canonUrl.pathname) return;
  canonEl.setAttribute('href', `${canonUrl.href}.html`);

```

Reply:
Good point! I tried to commit your change from the GUI, but it mangled the code, I've pushed it from my local